Clear up Highlight sound settings

Nerixyz opened the pull request:
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

As discussed in #1540, this clears up the highlight settings for sounds:

![](https://user-images.githubusercontent.com/19953266/204150044-88775fe6-e4f9-4493-9863-9010ecdfebc6.png)

<details><summary>Without custom default sound</summary>

![firefox_2022-11-27_18-14-59](https://user-images.githubusercontent.com/19953266/204150063-ad725c5b-a368-4b73-af2e-07318b00b381.png)
</details>

Closes #1540.

kornes commented:
suggestions:
- icon on clear button looks off, makes whole button kinda stands out too much. I would opt to remove it, or leave just icon, but then some variation of bin 🗑 would fit better i think.
- disable clear button when no custom sound is set
- maybe change current filename-only label, into input field with full file path, where user could paste/inspect path without clicking Change button. "Chatterino Ping" could be placeholder then.

8thony commented:
Things I noticed:
- If you have set a custom sound in the table before and then click to change the sound but hit the cancel button in the file browser, a ticked checkbox stays in the table. (Is this checkbox even necessary?)
- The table entry with no option to play a sound doesn't block the option to set a custom sound.

Nerixyz commented:
> * icon on clear button looks off, makes whole button kinda stands out too much. I would opt to remove it, or leave just icon, but then some variation of bin 🗑 would fit better i think.

Removed.

> * disable clear button when no custom sound is set

It's hidden now.

> * maybe change current filename-only label, into input field with full file path, where user could paste/inspect path without clicking Change button. "Chatterino Ping" could be placeholder then.

This doesn't really fit the other settings. A user can now click on the path to open a file.

> * If you have set a custom sound in the table before and then click to change the sound but hit the cancel button in the file browser, a ticked checkbox stays in the table. (Is this checkbox even necessary?)

I removed the checkbox since it wasn't used.

> * The table entry with no option to play a sound doesn't block the option to set a custom sound.

There were actually settings for the sound-urls. But they weren't used, so I removed them.

8thony commented:
If you set a sound with a long file name in the "Custom Sound" column and restart the program, the "Pattern" column becomes relatively narrow

If a sound with a long file name is set as "default sound", the settings window becomes wider

If you try this yourself, then make sure not to test the points above at the same time

pajlada commented on src/singletons/Settings.hpp:
```diff
@@ -271,7 +271,7 @@ class Settings : public ABSettings, public ConcurrentSettings
     /// Highlighting
     //    BoolSetting enableHighlights = {"/highlighting/enabled", true};
-    BoolSetting customHighlightSound = {"/highlighting/useCustomSound", false};
+    //    BoolSetting customHighlightSound = {"/highlighting/useCustomSound", false};
```
Remove the setting entirely

```suggestion
```

pajlada commented on src/messages/SharedMessageBuilder.cpp:
```diff
@@ -22,11 +22,12 @@ namespace {
-        // Use fallback sound when checkbox is not checked
-        // or custom file doesn't exist
-        if (getSettings()->customHighlightSound && fileExists)
+        // Use fallback sound when no custom sound is set
+        // or if the file doesn't exist.
```
Let's move this comment to above the function instead and describe what the function does

pajlada reviewed (changes requested):
Changing sounds doesn't update, it requires a restart

Nerixyz commented:
> If you set a sound with a long file name in the "Custom Sound" column and restart the program, the "Pattern" column becomes relatively narrow (Tested with 40 characters, including file extension)

Should be twice the width now, though this should really be changed after #3690.

> If a sound with a long file name is set as "default sound", the settings window becomes wider (Tested with 80 characters, including file extension)

Filenames are now truncated to 50 characters.

8thony commented:
> Should be twice the width now, though this should really be changed after #3690.

It doesn't change anything for me, but it seems to be a general issue that isn't unique to this settings page.

> Filenames are now truncated to 50 characters.

It's okay. Maybe it shouldn't be limited to a certain number but to the current size of the window. So making it wider reveals more of the name. I'll let you decide what is better.
